shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -169,20 +158,23 @@ private void validateAppend(Append append, Session session) {
                 "Bug in CommandEncoder.encode, block is too small.");
     }
 
-    private void breakFromAppend(ByteBuf out) {
+    private void breakFromAppend(ByteBuf data, ByteBuf pendingData, ByteBuf out) {
         if (bytesLeftInBlock != 0) {
-            writeMessage(new Padding(bytesLeftInBlock - TYPE_PLUS_LENGTH_SIZE), out);
+            if (data != null) {
```

Comment:
I don't understand this change. This is replacing a single line:

```
writeMessage(new Padding(bytesLeftInBlock - TYPE_PLUS_LENGTH_SIZE), out);
```

with:

```
            if (data != null) {
                 writeMessage(new PartialEvent(data), out);
            } else {
                writeMessage(new Padding(bytesLeftInBlock - TYPE_PLUS_LENGTH_SIZE), out);
            }
```

This seems to be unrelated to the tracking that is being added, please clarify.

Reply:
The appendbreak happens in the two conditions
1.  event size more than the append block size (max is 32K) ; then we write the PartialEvent  (if case) 
2.   current segment id or writer id changed on the netty channel, then we pad the remaining block size. (else case)